src/scripts/main.js

```diff
@@ -1,3 +1,430 @@
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    `;
+
+    td.innerHTML = '';
+    td.append(cellOffice);
+    cellOffice.focus();
+
+    cellOffice.onblur = () => {
+      const office = cellOffice.selectedIndex;
+
+      td.innerHTML = cellOffice[office].value;
+
+      if (cellOffice.selectedIndex === 0) {
+        td.innerHTML = editingArea.data;
+      }
+
+      pushNotification(
+        460,
+        10,
+        'Changed successfully!',
+        'Congrats! The office was successfully changed.',
+        'success');
+    };
+
+    cellOffice.onkeydown = (enter) => {
+      if (enter.key === 'Enter') {
+        cellOffice.onblur();
+      }
+    };
+  } else {
+    const cellInput = document.createElement('input');
+
+    cellInput.className = 'cell-input';
+    cellInput.innerHTML = td.innerHTML;
+    td.innerHTML = '';
+    td.append(cellInput);
+    cellInput.focus();
+
+    switch (td.cellIndex) {
+      case 0:
+      case 1:
+        cellInput.setAttribute('type', 'text');
+        break;
+
+      case 3:
+      case 4:
+        cellInput.setAttribute('type', 'number');
+        break;
+    }
+
+    cellInput.onblur = (e) => {
+      validate(td.cellIndex, cellInput);
+      editEnd(editingArea.elem, isCorrect);
+      changeIsCorrect();
+    };
+
+    cellInput.onkeydown = (enter) => {
+      if (enter.key === 'Enter') {
+        cellInput.onblur();
+      }
+    };
+  }
+};
+
+function validate(index, cell) {
+  switch (index) {
+    case 0:
+    case 1:
+      if (/\d/.test(cell.value)
+        || cell.value.trim().length < nameLength) {
+        isCorrect = false;
+
+        pushNotification(
+          580,
+          10,
+          'Enter a correct data!',
+          'This field must contain at least 4 letters and not include numbers.',
+          'error');
+      }
+
+      break;
+
+    case 3:
+      if (cell.value < 18
+        || cell.value > 90) {
+        if (cell.value === '') {
+          isCorrect = false;
+
+          return;
+        }
+
+        isCorrect = false;
+
+        pushNotification(
+          580,
+          10,
+          'Enter a correct age!',
+          'The age must be 18+ and less than 90.',
+          'error');
+      }
+
+      break;
+
+    case 4:
+      if (+cell.value <= 0) {
+        if (cell.value === '') {
+          isCorrect = false;
+
+          return;
+        }
+
+        isCorrect = false;
+
+        pushNotification(
+          580,
+          10,
+          'Enter a correct salary!',
+          'The salary must be more than 0.',
+          'error');
+      }
+
+      break;
+  }
+}
+
+function editEnd(td, isOk) {
+  if (isOk) {
+    switch (td.cellIndex) {
+      case 4:
```

Comment:
editing the current field every time leads to NaN:
![image](https://user-images.githubusercontent.com/86923089/219943311-4f9fb2fc-e316-43a1-b0c2-bc48a289acd6.png)
![image](https://user-images.githubusercontent.com/86923089/219943322-d78f4704-2fa5-44c8-b312-92d53aa6e862.png)